### PR TITLE
Add and fix docblocks in Json repositories about flags

### PR DESCRIPTION
**Pull request by tgalopin**

### CODE REVIEW

File: src/AbstractJsonRepository.php [hunk 1]
```diff
@@ -347,6 +347,8 @@ public function hasChildren($path)
      *
      * The keys of the returned array are Puli paths. Their order is undefined.
      *
+     * Flag `STOP_ON_FIRST` may be used to stop the search to the first result.
```
Comment (webmozart) [inline src/AbstractJsonRepository.php]: "The flag", "at the first result"

File: src/JsonRepository.php [hunk 2]
```diff
@@ -387,10 +387,15 @@ private function flatten(array $references)
...
-     * by passing a `$maxDepth` (see {@link getPathDepth()}).
+     * All references that contain directory paths may be traversed recursively and
+     * scanned for more paths matching the regular expression. This recursive
+     * traversal can be limited by passing a `$maxDepth` (see {@link getPathDepth()}).
+     * By default, this `$maxDepth` is equal to zero (no recursive scan).
+     *
+     * Flag `STOP_ON_FIRST` may be used to stop the search to the first result.
+     *
+     * Flag `NO_SEARCH_FILESYSTEM` may be used to not check the filesystem for the
+     * found results.
```
Comment (webmozart) [inline src/JsonRepository.php]: for whether the found paths actually exist on the filesystem

**Comment by tgalopin** [comment]
Done :)

File: src/AbstractJsonRepository.php [hunk 3]
```diff
@@ -44,7 +44,7 @@
 abstract class AbstractJsonRepository extends AbstractEditableRepository implements LoggerAwareInterface
 {
     /**
-     * Flag: Whether to stop after the first result.
+     * The flag: Whether to stop after the first result.
```
Comment (webmozart) [inline src/AbstractJsonRepository.php]: I guess this is a search replace mistake? :)

**Comment by tgalopin** [comment]
It is :) I change this

**Comment by tgalopin** [comment]
I did the change.

**Comment by tgalopin** [comment]
I think this is mergable :) .

File: src/AbstractJsonRepository.php [hunk 4]
```diff
@@ -44,7 +44,7 @@
 abstract class AbstractJsonRepository extends AbstractEditableRepository implements LoggerAwareInterface
 {
     /**
-     * Flag: Whether to stop after the first result.
```
Comment (webmozart) [inline src/AbstractJsonRepository.php]: Can you leave the "Flag: " in?

File: src/JsonRepository.php [hunk 5]
```diff
@@ -70,28 +70,28 @@
 class JsonRepository extends AbstractJsonRepository implements EditableRepository
 {
     /**
-     * Flag: Don't search the contents of mapped directories for matching paths.
```
Comment (webmozart) [inline src/JsonRepository.php]: Here too

**Comment by tgalopin** [comment]
Updated

**Comment by webmozart** [comment]
Thanks! :)

**Pull request merged**